src/job.rs

```diff
@@ -511,11 +511,28 @@ impl Job {
         }
 
         let msg = job_try!(Message::load_from_db(context, MsgId::new(self.foreign_id)).await);
-        let dest_folder = context.get_config(Config::ConfiguredMvboxFolder).await;
+        let server_folder = msg.server_folder.as_ref().unwrap();
```

Comment:
Since you are moving this code out of `if`, please remove the `unwrap`. Return a meaningful error if `server_folder` is `None` to make sure the app will not crash.